### PR TITLE
Shulk Changes

### PR DESCRIPTION
**Pull request by H3ra-57**
[hdr-assets-dev.zip](https://github.com/HDR-Development/HewDraw-Remix/files/14653969/hdr-assets-dev.zip)

Big thanks to Silent and Mokl for the asset side of the PR!

### General
- [/] Dash Speed: 1.732 → 1.6
- [/] Max Run Speed: 1.6 → 1.7
- [$] Given a new victory animation

### Jab 2
- [+] Damage: 1.5% → 2.5%

### Jab 3
- [+] Damage: 5% → 6%

### Dash Attack
- [/] Hitbox positioning/sizing reverted to vanilla
- [+] FAF: F41 → F38

### Forward Tilt
- [+] Hitbox added on arm (identical stats to sweetspot)
- [+] Startup: F10 → F9

### Up Tilt
- [-] Startup: F9 → F10

### Down Tilt
- [/] Hitbox ID 2 position/sizing reverted to vanilla
- [+] Startup: F10 → F8

### Forward Smash
- Hit 1
  - [+] Damage: 5.5% → 6%
- Hit 2
  - [+] Damage: 13.5%/11.5% → 14.5%/12.5%

### Up Smash
- Hit 1
  - [+] Damage: 4.5% → 5%
- Hit 2
  - [+] Damage: 13.5% → 15%

### Down Smash
- (!) Given a new animation, based on Monado Rage from Xenoblade 3
  - Hitbox Duration: F16-F18
  - Damage (Sweet/Sour): 14.5%/11.5%
  - Angle: 75/65/361
  - BKB: 50
  - KBG: 90
  - Hitbox Size: 4.3u/4.4u/4.5u/4.1u/3.8u
  - FAF: F53

### Neutral Air
- [+] Startup: F14 → F12
- [/] Duration: F14-F35 → F12-F28 (Attack animates faster)
- [+] Damage: 8.5%/7.5% → 10%/8.5%
- [/] BKB: 55/45 → 50/42
- [+] Landing Lag: 10F → 8F

### Forward Air
- [+] Startup: F14 → F12
- [+] Damage: 8%/6.5% → 10%/8.5%

### Back Air
- [+] Startup: F18 → F16
- [+] Damage: 9.5%/14.5% → 10.5%/15.5%
- [+] Landing Lag: 14F → 12F

### Up Air
- [+] Hit 2 Startup: F19 → F16
- [+] Hit 2 Damage [Early, Late]: 11%/8.5%, 7.5%/4.5% → 12%/9.5%, 8%/5%
- [-] Landing Lag: 9F → 11F

### Down Air
- [+] Hit 2 Startup: F20 → F19

### Monado Arts
- Art Wheel
  - [-] FAF (cancel with no art chosen): F1 → F4
  - [-] Intangibility end (selecting an art): F15 → F5
- Attributes
  - Jump
    - [+] Time active: 6s → 12s
    - [+] Time unavailable: 18s → 15s
    - [-] Jump height mul    : 2.0x → 1.90x
    - [-] Air speed mul      : 1.4x → 1.30x
    - [-] Fall speed mul     : 1.4x → 1.35x
    - [-] Fast fall speed mul: 1.4x → 1.35x
    - [+] Damage taken mul   : 1.3x → 1.22x
  - Speed
    - [+] Time active: 8s → 12s
    - [+] Time unavailable: 16s → 15s
    - [-] Dash speed mul  : 1.7x → 1.60x
    - [-] Run speed mul   : 1.7x → 1.60x
    - [/] Friction mul    : 1.4x → 1.30x
    - [-] Jump height mul : 0.9x → 0.85x
  - Shield
    - [$] Renamed to Armour
    - [+] Time active: 6s → 12s
    - [+] Time unavailable: 18s → 15s
    - [-] Damage taken mul: 0.6x → 0.7x
    - [+] Damage dealt mul: 0.5x → 0.6x
    - [+] Walk speed mul  : 0.6x → 0.7x
    - [+] Run speed mul   : 0.6x → 0.7x
    - [+] Air speed mul   : 0.6x → 0.7x
  - Buster
    - [$] Renamed to Enchant
    - [+] Time active: 10s → 12s
    - [-] Time unavailable: 14s → 15s
    - [+] Damage taken mul   : 1.30x → 1.20x
    - [+] Knockback dealt mul: 0.65x → 0.68x
  - Smash
    - [$] Renamed to Rage
    - [+] Time active: 6s → 12s
    - [+] Time unavailable: 16s → 15s
    - [-] Knockback dealt mul: 1.25x → 1.2x
    - [+] Damage dealt mul   : 0.30x → 0.5x
- Arts Cancelling
  - [-] Removed ability to change monado arts during hitstun
  - (!) In its place, Shulk can now change monado arts when hitting jab 3 and all tilts and aerials. Entering the Art Wheel in this state will cancel the attack.

### Back Slash → Heavy Rush
- (!) Changed to Heavy Rush from Xenoblade 3, a dashing move with different effects based on the active monado art
  - Hitbox Duration: F21-F28
  - Shield has 20% heavy armor from F5-F28
  - Damage (Sweet/Sour) [Normal, Buster, Smash]: 12.0%/9.0%, 13.0%/9.7%, 10.2%/7.7%
  - Angle [Normal, Jump]: 361, 80
  - BKB [Normal, Smash]: 70, 75
  - KBG [Normal, Jump, Smash]: 80, 60, 85
  - Hitbox Size (Sweet/Sour): 4.0u/3.5u
  - FAF: F58
  - Ledge cancellable F30+
  - Land cancellable F40+ 

### Vision
- [+] Counterattack made unblockable

### CODE REVIEW

File: fighters/shulk/src/opff.rs [hunk 1]
```diff
@@ -68,11 +51,44 @@ unsafe fn fastfall_specials(fighter: &mut L2CFighterCommon) {
...
+        Hash40::new("attack_hi3"),
+        Hash40::new("attack_lw3"),
+        Hash40::new("attack_air_n"),
+        Hash40::new("attack_air_f"),
+        Hash40::new("attack_air_b"),
+        Hash40::new("attack_air_hi"),
+        Hash40::new("attack_air_lw") ]) {
+        if AttackModule::is_infliction_status(fighter.module_accessor, *COLLISION_KIND_MASK_HIT | *COLLISION_KIND_MASK_SHIELD) {
+            println!("beat");
+            VarModule::on_flag(fighter.object(), vars::shulk::instance::MONADO_BEAT);
```
Comment (WuBoytH) [inline fighters/shulk/src/opff.rs]: Please make this a status variable instead, and remove the second half of the function that turns off the flag

Comment (WuBoytH) [reply fighters/shulk/src/opff.rs]: oh and the println I guess

File: fighters/shulk/src/acmd/aerials.rs [hunk 2]
```diff
@@ -104,22 +109,18 @@ unsafe extern "C" fn shulk_attack_air_hi_game(fighter: &mut L2CAgentBase) {
...
     }
     frame(lua_state, 5.0);
     if is_excute(fighter) {
         WorkModule::on_flag(boma, *FIGHTER_STATUS_ATTACK_AIR_FLAG_ENABLE_LANDING);
     }
     frame(lua_state, 12.0);
-    if is_excute(fighter) {
-        FT_MOTION_RATE(fighter, 1.000);
-    }
+    FT_MOTION_RATE(fighter, 1.000);
```
Comment (WuBoytH) [inline fighters/shulk/src/acmd/aerials.rs:121]: Freed from the clutches of is_excute...

**Review by WuBoytH — DISMISSED** [review]

File: fighters/shulk/src/opff.rs (hunk 1 above)
Comment (H3ra-57) [reply fighters/shulk/src/opff.rs]: done. however, making this a status variable makes the effects on a successful arts cancel not play.

Comment (WuBoytH) [reply fighters/shulk/src/opff.rs]: That can probably be resolved via modifying the monado art activation pre status and setting the variable again

**Pull request merged**